### DISCUSSION TITLE
how to create lsp workspace and add files to it?

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#2120] [Q&A] [answered]
Not lazyvim specific question. But After a lot of search work I cant find a straight answer. 

I need 2 things. 
1. Find all references of method or variable in project (multi-files)
2. Go to definition with command `gd` across file. 

how to make this?

**Comment by kobzar** [comment, marked as answer]
All of this inside your LSP server
Just check if LSP is working for your filetype and use 
gd
gr 
for this
Also, put cursor to selected method and press g
wait a second and you see all variants that you have
